include/openssl/dsa.h

```diff
@@ -130,53 +143,56 @@ DEPRECATEDIN_0_9_8(DSA *DSA_generate_parameters(int bits,
                                                 void *cb_arg))
 
 /* New version */
-int DSA_generate_parameters_ex(DSA *dsa, int bits,
-                               const unsigned char *seed, int seed_len,
-                               int *counter_ret, unsigned long *h_ret,
-                               BN_GENCB *cb);
+DEPRECATEDIN_3_0(int DSA_generate_parameters_ex(DSA *dsa, int bits,
+                                                const unsigned char *seed,
+                                                int seed_len, int *counter_ret,
+                                                unsigned long *h_ret,
+                                                BN_GENCB *cb))
 
-int DSA_generate_key(DSA *a);
+DEPRECATEDIN_3_0(int DSA_generate_key(DSA *a))
 
 int DSAparams_print(BIO *bp, const DSA *x);
 int DSA_print(BIO *bp, const DSA *x, int off);
-# ifndef OPENSSL_NO_STDIO
+#  ifndef OPENSSL_NO_STDIO
 int DSAparams_print_fp(FILE *fp, const DSA *x);
 int DSA_print_fp(FILE *bp, const DSA *x, int off);
-# endif
+#  endif
```

Comment:
Is there a reason you don't deprecate the printing functions?

Reply:
It seemed like they were a good fit with the serialisers.